Common/include/basic_types/ad_structure.hpp

```diff
@@ -299,7 +299,7 @@ namespace AD{
 
   FORCEINLINE void RegisterOutput(su2double& data) {AD::getGlobalTape().registerOutput(data);}
 
-  FORCEINLINE void ResetInput(su2double &data) {data.getGradientData() = su2double::GradientData();}
+  FORCEINLINE void ResetInput(su2double &data) {data = data.getValue();}
```

Comment:
👍 now it is fully equivalent with what is done during CLEAR_INDICES recording.

Reply:
at least from the standpoint that you clear the index by writing to the primal variable

Reply:
Thanks for adding this already @pcarruscag.

Reply:
The fastest keyboard in the west 🤠